src/libraries/System.Web.HttpUtility/src/System/Web/Util/HttpEncoder.cs

```diff
@@ -258,7 +258,9 @@ internal static string JavaScriptStringEncode(string? value)
                 return null;
             }
 
-            UrlDecoder helper = new UrlDecoder(count, encoding);
+            UrlDecoder helper = count < 256
+                ? new UrlDecoder(stackalloc char[count], stackalloc byte[count], encoding)
```

Comment:
if I remember correctly, the recommendation is not to use the actual count for stackalloc then, but the max value (256 in your case) and slice it to size.
I'm also not sure if 256 is the correct limit, as this would  mean 255*3 bytes on the stack, this might be a bit much for certain environments.
According to https://www.researchgate.net/publication/360254493_An_Adversarial_Attack_Analysis_on_Malicious_Advertisement_URL_Detection_Framework the average length of an url is 44 chars, so you might just use 128 as your limit.

Reply:
Thanks for the suggestion about the constant. I will push changes, just waiting for suggestion from owners what is their prefer limit. I saw several other places in runtime repo where they choose 256 as limit and were not afraid that it is too much.

Reply:
256 seems fine here, we mostly tend to stay <= 1 KB total.
Should be just using the constant directly as suggested though.

Reply:
One problem with the constant size is that it significantly increases stack consumption. so having this method on call stack is automatically +512 bytes esp when input is mosly small like 32 bytes.

The non-constant stackalloc is a bit expensive due to extra instructions to do stack/page probing. It is also a lot more expensive when we need to zero it, but it's not the case for BCL where we mostly have SkipLocalsInit.